### PR TITLE
Fix responsive styles when using custom breakpoints

### PR DESCRIPTION
**Pull request by craigpalermo**
This fixes a bug when passing `customBreakpoints` to `<ThemeProvider>` and using responsive props arrays. Some changes internally to Styled System 5 seem to have caused this behavior change, but using Styled System's `compose` function fixes it.

### CODE REVIEW

**Review by pherrera167 — APPROVED** [review]
LGTM

File: packages/modal/src/Modal.spec.js [hunk 1]
```diff
@@ -77,12 +77,12 @@ describe('Modal', () => {
 
     expect(getByText('Content')).toBeTruthy()
     expect(dialogContent).toHaveStyle(`
-      height: 100vh;
-      width: 100vw;
+      height: 100%;
```
Comment (sdalonzo) [inline packages/modal/src/Modal.spec.js:80]: ?

File: packages/core/src/utils/createTheme.ts [hunk 2]
```diff
@@ -38,6 +39,8 @@ const createTheme = (theme = {}, customBreakpoints = null, existingTheme = defau
     colorStyles: createColorStyles(finalMergedTheme),
     textStyles: createTextStyles(finalMergedTheme),
   }
+  // console.log(t)
```
Comment (sdalonzo) [inline packages/core/src/utils/createTheme.ts:42]: console

File: packages/modal/src/Modal.js [hunk 3]
```diff
@@ -76,6 +73,12 @@ const Dialog = styled(DialogContent)`
     max-height: none;
   `}
   ${getAnimation}
+
+  ${(props) => {
+    const c = compose(color, width, height)(props)
+    console.log(c)
```
Comment (sdalonzo) [inline packages/modal/src/Modal.js:79]: console

File: packages/core/src/utils/createTheme.ts (hunk 2 above)
Comment (gbellini911) [reply packages/core/src/utils/createTheme.ts]: ooooOooOoo it was for your console log.

**Review by sdalonzo — APPROVED** [review]

**Pull request merged**